log/sequencer.go

```diff
@@ -11,13 +11,15 @@ import (
 	"github.com/google/trillian/merkle"
 	"github.com/google/trillian/storage"
 	"github.com/google/trillian/util"
+	"golang.org/x/net/context"
 )
 
-// Sequencer instances are responsible for integrating new leaves into a log.
+// Sequencer instances are responsible for integrating new leaves into a single log.
 // Leaves will be assigned unique sequence numbers when they are processed.
 // There is no strong ordering guarantee but in general entries will be processed
 // in order of submission to the log.
 type Sequencer struct {
+	ctx        context.Context
```

Comment:
Done, although the net result makes me wonder whether a better approach might be to drop the `Context` key:value and instead add a `MapID()` method to the various log storage `interface`s, analogous to `MapStorage.MapID()`.  Then the diagnostics could just get the ID back from the storage instance.  Preferences?